test/box-luatest/gh_10349_functional_index_without_sandbox_test.lua

```diff
@@ -0,0 +1,71 @@
+local server = require('luatest.server')
+local t = require('luatest')
+
+local g = t.group()
+
+g.before_all(function(cg)
+    cg.server = server:new()
+    cg.server:start()
+end)
+
+g.after_all(function(cg)
+    cg.server:drop()
+end)
+
+g.after_each(function(cg)
+    cg.server:exec(function()
+        if box.space.test ~= nil then
+            box.space.test:drop()
+        end
+        box.schema.func.drop('test', {if_exists = true})
+    end)
+end)
+
+g.test_func_index_without_sandbox = function(cg)
+    cg.server:exec(function()
+        local msgpack = require('msgpack')
+        local varbinary = require('varbinary')
+        box.schema.func.create('test', {
+            is_deterministic = true,
+            body = [[function(tuple)
+                local msgpack = require('msgpack')
+                local varbinary = require('varbinary')
+                return {varbinary.new(msgpack.encode(tuple.value))}
+            end]]
+        })
+        local s = box.schema.space.create('test', {
+            format = {
+                {'key', 'unsigned'},
+                {'value', 'any'},
+            },
+        })
+        s:create_index('primary')
+        s:create_index('value', {
+            func = 'test',
+            parts = {{1, 'varbinary'}},
+        })
+        s:insert({1, {foo = 'bar'}})
+        s:insert({2, {fuzz = 'buzz'}})
+        local function get(value)
+            local v = varbinary.new(msgpack.encode(value))
+            return box.space.test.index.value:get(v)
+        end
+        t.assert_equals(get({foo = 'bar'}), {1, {foo = 'bar'}})
+        t.assert_equals(get({fuzz = 'buzz'}), {2, {fuzz = 'buzz'}})
+        t.assert_equals(get({1, 2, 3}), nil)
+        box.snapshot()
+        s:insert({3, {1, 2, 3}})
+    end)
+    cg.server:restart()
+    cg.server:exec(function()
+        local msgpack = require('msgpack')
+        local varbinary = require('varbinary')
+        local function get(value)
+            local v = varbinary.new(msgpack.encode(value))
+            return box.space.test.index.value:get(v)
+        end
+        t.assert_equals(get({foo = 'bar'}), {1, {foo = 'bar'}})
+        t.assert_equals(get({fuzz = 'buzz'}), {2, {fuzz = 'buzz'}})
+        t.assert_equals(get({1, 2, 3}), {3, {1, 2, 3}})
+    end)
```

Comment:
What is the idea behind this part of the test case?

I understood it this way: the part above checks that a functional index works correctly for values written during a lifetime of the given process. The quoted part of the test splits writes into a space with a functional index into two parts: the first is written into a snapshot, the second is written into WAL. This way it checks that the functional index is operational for values read from a snapshot or WAL, not only for ones that are written by the process during its lifespan.

OTOH, I don't know, why it needs attention. I guess it is possible that local recovery may construct functional index in a separate code path that has different checks, and the test verifies that both paths allows a non-sandboxed function.

However, it is just guess. I would like if you drop a comment regarding an idea behind it.

Reply:
I don't know why I added this. Just in case, I guess. Functional indexes are created in the same code path both during recovery and normal operation. We don't have any extra checks that would differ so the code is fully covered by the pre-restart test part. I dropped this confusing part from the test.

Reply:
OK, I don't mind.